docs/src/whatsnew.rst

```diff
@@ -8,6 +8,11 @@ This document outlines features and improvements from each release.
    are for non-production testing and evaluation, and may include
    changes to the API.
 
+v1.14.0 - May 28, 2025
```

Comment:
You can leave this, but for future reference, we try to do releases in their own commit. We have to update the generated documentation as well when we do a release, which reminds me, you should add these API changes to `docs/src/api.rst`.